onnxruntime/core/framework/session_state.h

```diff
@@ -280,6 +284,8 @@ class SessionState {
   const logging::Logger* logger_ = nullptr;
   profiling::Profiler* profiler_ = nullptr;
 
+  bool use_deterministic_compute_;
```

Comment:
I still feel this should be per-session option, see SessionOptions. 
When we construct the session, we should determine this flag... 

+ Jesse and Cheng for opinions

Reply:
good point. I moved the flag from training_parameter to session_option so that both inference and training sessions can choose to use deterministic compute. Regarding having a flag in session_state, my understanding is that session_state is the only thing that can be passed to the kernel. 

BTW, this flag will not change during training.

Reply:
@jessebenson @souptc , any opinions?